docs/start/authentication/graphql-storefront.mdx

```diff
@@ -66,17 +66,226 @@ content-type: application/json
+                },
+                "favoriteColor": {
+                    "entityId": 456,
+                    "value": blue
+                }
+            }
+        }
+    }
+}
+```
+</Tab>
+</Tabs>
+
+
 ### Using auto-generated tokens on Stencil storefronts
 
 On Stencil storefronts, you can access a token at render time and pass the token to client-side code using the `{{settings.storefront_api.token}}` Handlebars property. This auto-generated token has an expiry period of 24-48 hours and will rotate before expiration.
 
+
 ## Customer impersonation tokens
 
-You can use customer impersonation tokens to authenticate requests to the GraphQL Storefront API in server-to-server and headless interactions. The API will reject any customer impersonation token-authenticated requests that originate from a web browser and are not proxied. Use these tokens in NextJS applications; NextJS proxies requests by default.
+<Callout type="warning">
+Because of the flexible use of the customer impersonation token, BigCommerce recommends using a more secure option like the [customer access token](#customer-access-tokens). Customer impersonation tokens allow you to access any customer by simply changing the `X-Bc-Customer-Id` value in the header for server-to-server communications. However, the customer access token is tied directly to a specific customer as part of the token payload providing a more secure experience. In addition, customer access tokens have the following added benefits:
+* A customer access token is unique to an individual user's account.
+* A customer access token allows users to act on different systems.
+* A customer access token becomes invalid on all devices when you log out on a single device.
+</Callout>
+
+You can use customer impersonation tokens to authenticate requests to the GraphQL Storefront API in server-to-server and headless interactions. The API will reject any customer impersonation token-authenticated requests that originate from a web browser and are not proxied. Use these tokens in NextJS applications; NextJS proxies requests by default. 
 
-Customer impersonation token-authenticated requests can query store information for any customer by passing the customer ID using the `X-Bc-Customer-Id` header. Information that you request will be specific to the subject customer, including product pricing and availability, customer account details, and more.
+Customer impersonation token-authenticated requests will be specific to the customer, including product pricing and availability, customer account details, and more.
 
-It is not necessary to generate a new token for each customer ID. You may use a single token at any given time for your application, and specify a distinct customer ID for each set of requests.
+Generating a new token for each customer ID is optional. You may use a single token at any given time for your application and specify a distinct customer ID for each set of requests. 
```

Comment:
I wouldn't even call it optional, there's no good reason to do it.

I would say "is not necessary".